overlord/ifacestate/helpers.go

```diff
@@ -297,18 +297,19 @@ func CheckInterfaces(st *state.State, snapInfo *snap.Info) error {
 	// XXX: AddImplicitSlots is really a brittle interface
 	snap.AddImplicitSlots(snapInfo)
 
+	if snapInfo.SnapID == "" {
+		// no SnapID means --dangerous was given, so skip interface checks
```

Comment:
Yeah, it's probably a good idea to error. We're not really doing anything useful when it comes from the store at the moment.